### ISSUE TITLE
Privacy Manifest

### ISSUE THREAD

**Issue by lambda123254** [InderKumarRathore/DeviceGuru#120]
In my understanding, libraries which use one of this function listed in [Apple Doc](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api) should add Privacy Manifest to their library.

<img width="1007" alt="image" src="https://github.com/InderKumarRathore/DeviceGuru/assets/103159356/653800e5-256c-4517-b958-9baaeb021e02">

As far as my concerned, this library use User Defaults for read and write purpose. So any plan about this?
Correct me if I'm wrong, I'm still learning about this as well.

**Comment by ilanahaddad** [comment]
Hi, wondering the same here if this will be addressed and a new version will be released including the manifest?

**Comment by InderKumarRathore** [comment]
Added changes into `10.0.6`, please check if this works for you. I created new PR to fix some issues while pushing to cocoapods

https://github.com/InderKumarRathore/DeviceGuru/pull/123

**Comment by gfoldv** [comment]
The xcode generated privacy report states:

`Missing an expected key: 'NSPrivacyCollectedDataTypes'`

**Comment by InderKumarRathore** [comment]
Can you please provide steps to reproduce this

**Comment by gfoldv** [comment]
﻿﻿﻿﻿https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4239187

The PrivacyInfo.xcprivacy file should look like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
  <key>NSPrivacyTracking</key>
  <false/>
  <key>NSPrivacyCollectedDataTypes</key>
  <array/>
  <key>NSPrivacyTrackingDomains</key>
  <array/>
  <key>NSPrivacyAccessedAPITypes</key>
  <array>
    <dict>
      <key>NSPrivacyAccessedAPIType</key>
      <string>NSPrivacyAccessedAPICategoryUserDefaults</string>
      <key>NSPrivacyAccessedAPITypeReasons</key>
      <array>
        <string>CA92.1</string>
      </array>
    </dict>
  </array>
</dict>
</plist>
```

**Comment by mrakhal** [comment]
Hi @InderKumarRathore to reproduce Missing an expected key: 'NSPrivacyCollectedDataTypes' you can generate privacy manifest report via archive, then will shown the result.

I already fix this issues, Thank you 🚀
#124 Update Privacy Manifest Requirements

**Comment by mrakhal** [comment]
> ﻿﻿﻿﻿https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4239187
> 
> The PrivacyInfo.xcprivacy file should look like this:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
>   <key>NSPrivacyTracking</key>
>   <false/>
>   <key>NSPrivacyCollectedDataTypes</key>
>   <array/>
>   <key>NSPrivacyTrackingDomains</key>
>   <array/>
>   <key>NSPrivacyAccessedAPITypes</key>
>   <array>
>     <dict>
>       <key>NSPrivacyAccessedAPIType</key>
>       <string>NSPrivacyAccessedAPICategoryUserDefaults</string>
>       <key>NSPrivacyAccessedAPITypeReasons</key>
>       <array>
>         <string>CA92.1</string>
>       </array>
>     </dict>
>   </array>
> </dict>
> </plist>
> ```

Hi @gfoldv ,The provided code only resolves the warning 'Missing an expected key: 'NSPrivacyCollectedDataTypes'. However, if you generate a privacy manifest report, it won't display the DeviceGuru privacy manifest because the data inside the array `NSPrivacyCollectedDataTypes` needs to be defined, CMIIW.

**Comment by lixiang1994** [comment]
Missing an expected key: 'NSPrivacyCollectedDataTypes'

**Comment by lixiang1994** [comment]
@InderKumarRathore

**Comment by InderKumarRathore** [comment]
Thanks @mrakhal for creating a PR, I have merged it and the new changes should be available in `10.0.7`
